pylmod/tests/test_membership.py

```diff
@@ -2,23 +2,172 @@
+        """Handle course id getting call"""
+        body = self.COURSE_DATA
+        if not send_data:
+            del body['response']
+        httpretty.register_uri(
+            httpretty.GET,
+            '{0}courseguide/course?uuid={1}'.format(
+                self.MEMBERSHIP_REGISTER_BASE,
+                self.CUUID,
+            ),
+            body=json.dumps(body)
+        )
+
+    def _register_get_staff(self):
+        """Handle staff getting API call"""
+        staff_data = self.STAFF_BODY
+        httpretty.register_uri(
+            httpretty.GET,
+            '{0}courseguide/course/{1}/staff'.format(
+                self.MEMBERSHIP_REGISTER_BASE,
+                self.COURSE_ID,
+            ),
+            body=json.dumps(staff_data)
+        )
+
     def test_constructor(self):
         """Verify constructor does as expected i.e. append URLBASE with
         ``service/membership/``
         """
         # Strip off base URL to make sure it comes back
         urlbase = self.URLBASE[:-1]
-        test_membership = Membership(self.CERT, urlbase)
+        test_base = Membership(self.CERT, cuuid=None, urlbase=urlbase)
         self.assertEqual(
-            test_membership.urlbase,
+            test_base.urlbase,
             self.URLBASE + 'service/membership/'
         )
-        self.assertEqual(test_membership._session.cert, self.CERT)
-        self.assertIsNone(test_membership.gradebookid)
+        self.assertEqual(test_base._session.cert, self.CERT)
+        self.assertIsNone(test_base.course_id)
+
+    @httpretty.activate
+    def test_constructor_with_cuuid(self):
+        """Verify we can construct with CUUID and properly add
+        self.course_id
+        """
+        self._register_get_course_id()
+        test_base = Membership(self.CERT, self.URLBASE, self.CUUID)
+        self.assertEqual(test_base.course_id, self.COURSE_ID)
+
+        # Also verify we made an API call
+        last_request = httpretty.last_request()
+        self.assertEqual(last_request.querystring, dict(uuid=[self.CUUID]))
+
+    @httpretty.activate
+    def test_get_staff(self):
+        """Verify that we can get staff roster as expected."""
+        self._register_get_staff()
+        test_membership = Membership(self.CERT, self.URLBASE)
+        staff_list = test_membership.get_staff(course_id=self.COURSE_ID)
+        self.maxDiff = None
+        self.assertEqual(staff_list, self.STAFF_BODY['response']['docs'])
+
+    @httpretty.activate
+    def test_get_course_id(self):
+        """Verify that we can the course id by providing the course uuid."""
+        self._register_get_course_id()
+        test_base = Membership(self.CERT, self.URLBASE)
+        course_id = test_base.get_course_id(self.CUUID)
+        self.assertEqual(course_id, self.COURSE_ID)
+
+        # Remove data and assert exception raised
+        self._register_get_course_id(False)
+        with self.assertRaises(PyLmodUnexpectedData):
```

Comment:
Paired with my first suggestion, I would mock the log and raise first a key error and second a type error and assert the correct log message is given for each.